packages/nx/src/utils/create-package-json.ts

```diff
@@ -1,10 +1,34 @@
 import { readJsonFile } from './fileutils';
 import { sortObjectByKeys } from './object-sort';
-import { ProjectGraph } from '../config/project-graph';
+import {
+  FileData,
+  ProjectGraph,
+  ProjectGraphDependency,
+} from '../config/project-graph';
 import { PackageJson } from './package-json';
 import { existsSync } from 'fs';
 import { workspaceRoot } from './workspace-root';
 
+// all dependencies from those files will be ignored
+const devOnlyFiles = [
+  // test files
+  '.spec.ts',
+  '.test.ts',
+  '.spec.tsx',
+  '.test.tsx',
+  '.spec.js',
+  '.test.js',
+  '.spec.jsx',
+  '.test.jsx',
+  // jest configs
+  'jest.config.ts',
+  'jest.preset.js',
+  // eslint config
+  '.eslintrc.js',
+  '.eslintrc.cjs',
+  'eslint.config.js',
+];
```

Comment:
Can we read the supported patterns from inputs and exclude files that don't match instead of hard coding this?